src/arch/x86/smbios.c

```diff
@@ -277,76 +277,38 @@ static int create_smbios_type17_for_dimm(struct dimm_info *dimm,
 	return smbios_full_table_len(&t->header, t->eos);
 }
 
-#define VERSION_VPD "firmware_version"
-static const char *vpd_get_bios_version(void)
+static const char *get_bios_framework_string(void)
 {
-	int size;
-	const char *s;
-	char *version;
-
-	s = vpd_find(VERSION_VPD, &size, VPD_RO);
-	if (!s) {
-		printk(BIOS_ERR, "Find version from VPD %s failed\n", VERSION_VPD);
-		return NULL;
-	}
-
-	version = malloc(size + 1);
-	if (!version) {
-		printk(BIOS_ERR, "Failed to malloc %d bytes for VPD version\n", size + 1);
-		return NULL;
-	}
-	memcpy(version, s, size);
-	version[size] = '\0';
-	printk(BIOS_DEBUG, "Firmware version %s from VPD %s\n", version, VERSION_VPD);
-	return version;
-}
+	if (CONFIG(PAYLOAD_SEABIOS))
+		return "coreboot+SeaBIOS";
 
-static const char *get_bios_version(void)
-{
-	const char *s;
-
-#define SPACES \
-	"                                                                  "
+	if (CONFIG(PAYLOAD_EDK2))
+		return "coreboot+UEFI";
 
-	if (CONFIG(CHROMEOS))
-		return SPACES;
-
-	if (CONFIG(VPD_SMBIOS_VERSION)) {
-		s = vpd_get_bios_version();
-		if (s != NULL)
-			return s;
-	}
-
-	s = smbios_mainboard_bios_version();
-	if (s != NULL)
-		return s;
-
-	if (strlen(CONFIG_LOCALVERSION) != 0) {
-		printk(BIOS_DEBUG, "BIOS version set to CONFIG_LOCALVERSION: '%s'\n",
-			CONFIG_LOCALVERSION);
-		return CONFIG_LOCALVERSION;
-	}
-
-	printk(BIOS_DEBUG, "SMBIOS firmware version is set to coreboot_version: '%s'\n",
-		coreboot_version);
-	return coreboot_version;
+	return "coreboot";
 }
 
 static int smbios_write_type0(unsigned long *current, int handle)
 {
 	struct smbios_type0 *t = smbios_carve_table(*current, SMBIOS_BIOS_INFORMATION,
 						    sizeof(*t), handle);
 
-	t->vendor = smbios_add_string(t->eos, "coreboot");
+	char bversion[100];
+
+	t->vendor = smbios_add_string(t->eos, "3mdeb");
 	t->bios_release_date = smbios_add_string(t->eos, coreboot_dmi_date);
 
+	snprintf(bversion, sizeof(bversion), "Dasharo (%s) %s",
+		get_bios_framework_string(), dasharo_version);
```

Comment:
```suggestion
		get_bios_framework_string(), CONFIG_LOCALVERSION);
```

Not sure if you noticed, but all those dasharo_version and dasharo_{minor, major, patch}_revision fields are nowhere used. If we are to sue the LOCALVERSION anyways we can simply use it directly.

Reply:
Ah, the major and minor versions were previously used in `system_bios_{major,minor)_release`, but that apparently that was lost, will fix

Reply:
Added. `dasharo_patch_revision` remains unused